### ISSUE TITLE
idrac facts

### ISSUE THREAD

**Issue by feisa** [dell/redfish-ansible-module#89]
receiving the following when trying to gather idrac_redfish_facts

{"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "msg": "Invalid Command 'GetManagerAttributes'. Valid Commands = []"}

- name: Get Manager attributes with a default of 20 seconds
  idrac_redfish_command:
    category: Manager
    command: GetManagerAttributes
    baseuri: "{{ baseuri }}"
    username: "{{ username }}"
    password: "{{ password }}"
    timeout: 20

**Comment by jose-delarosa** [comment]
Just tested on a 13G and 14G server and can not reproduce. I get the expected error on 13G:

`fatal: [r630]: FAILED! => {"changed": false, "msg": "HTTP Error 404 on GET request to 'https://idrac-r630/redfish/v1/Managers/iDRAC.Embedded.1/Attributes', extended message: 'Unable to complete the operation because the resource Attributes entered in not found.'"}
`

But on 14G all is well. Again, make sure you are using at least Ansible v2.8 from upstream.

**Comment by feisa** [comment]
HI, 

Ansible version 
ansible 2.9.0.dev0
  config file = /home/feisa/ansible-idrac/ansible.cfg
  configured module search path = [u'/usr/lib/python2.7/site-packages/ansible/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr  9 2019, 14:30:50) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

Role 

---
- name: Get Manager attributes with a default of 20 seconds
  idrac_redfish_command:
    category: Manager
    command: GetManagerAttributes
    baseuri: "{{ baseuri }}"
    username: "{{ user }}"
    password: "{{ password }}"
    timeout: 20

Error 

fatal: [idrac-ch1-002]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "baseuri": "idrac-ch1-002", 
            "category": "Manager", 
            "command": [
                "GetManagerAttributes"
            ], 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "timeout": 20, 
            "username": "root"
        }
    }, 
    "msg": "Invalid Command 'GetManagerAttributes'. Valid Commands = []"
}

**Comment by jose-delarosa** [comment]
What server? What iDRAC FW? I can't reproduce., not sure what else to do.

Also, when posting code or command output, use the code brackets (the little <> on top). It's hard to read output without it.

**Comment by billdodd** [comment]
>   idrac_redfish_command:
>   category: Manager
>   command: GetManagerAttributes
>   baseuri: "{{ baseuri }}"
>   username: "{{ user }}"
>   password: "{{ password }}"
>   timeout: 20
> 
> ...
>
> Error
> 
> ...
>
> "msg": "Invalid Command 'GetManagerAttributes'. Valid Commands = []"

`GetManagerAttributes` is in the `idrac_redfish_facts` module. Your playbook is using the `idrac_redfish_command` module.

**Comment by jose-delarosa** [comment]
Great catch @billdodd ! I missed that.

**Comment by feisa** [comment]
That works, I think the documentation needs to be updated because it says command

**Comment by jose-delarosa** [comment]
@feisa Yup I see that, thank you. I will submit a Pull Request, but I have to wait until https://github.com/ansible/ansible/pull/55536 is merged first.

**Comment by alessiodini** [comment]
Hello,
today the documentation is still wrong, please update it!!

**Comment by anupamaloke** [comment]
@alessiodini, the documentation is fixed in Ansible v2.10 which is still in development stage.
